### PR TITLE
[candi] change terraform version to 0.14.8

### PR DESCRIPTION
**Pull request by universal-itengineer**
## Description

Change terraform version from 0.13.8 to 0.14.8
## Why do we need it, and what problem does it solve?

This is required for the virtualization module so that virtual environments can be deployed using terraform-provider-kubernetes. This module can work with custom resources (vm, vmd, vmip, etc.) only with terraform starting from 0.14.8 version.
## What is the expected result?

All cloud environments must work as expected and cluster deployments across clouds must also work as expected.
## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi 
type: chore
summary: change terraform version to 0.14.8
impact: all cloud envirinment(clusters in AWS,GCP,Azure,Yandex-Cloud,OpenStack,VSphere)
impact_level: default 
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/aws'...

**Comment by universal-itengineer** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 7914941575 7914941575-1-con-1-25 /sys/deckhouse-oss/install:pr7513

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by universal-itengineer** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 7916416662 7916416662-2-con-1-25 /sys/deckhouse-oss/install:pr7513

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/gcp'...

**Comment by universal-itengineer** [comment]
/e2e/abort gcp;WithoutNAT;containerd;1.25 7917857708 7917857708-1-con-1-25 /sys/deckhouse-oss/install:pr7513

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by universal-itengineer** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 7929885872 7929885872-1-con-1-25 /sys/deckhouse-oss/install:pr7513

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/aws'...

**Comment by universal-itengineer** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 7932500721 7932500721-1-con-1-25 /sys/deckhouse-oss/install:pr7513

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by universal-itengineer** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 7935527029 7935527029-1-con-1-25 /sys/deckhouse-oss/install:pr7513

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by universal-itengineer** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 7935782068 7935782068-1-con-1-25 /sys/deckhouse-oss/install:pr7513

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by universal-itengineer** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 7957596960 7957596960-1-con-1-25 /sys/deckhouse-oss/install:pr7513

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've recognized your '/e2e/abort' command and started the workflow...

File: testing/cloud_layouts/script_eks.sh [hunk 1]
```diff
@@ -273,6 +273,21 @@ function cleanup() {
...
+    chown -R $USER_RUNNER_ID "$(pwd)/testing" || true
+    chown -R $USER_RUNNER_ID "/deckhouse/testing" || true
+    chown -R $USER_RUNNER_ID /tmp || true
+  else
+    chmod -f -R 777 "$(pwd)/testing" || true
+    chmod -f -R 777 "/deckhouse/testing" || true
+    chmod -f -R 777 /tmp || true
+  fi
+
+  echo "Rights and owner changed"
```
Comment (diafour) [inline testing/cloud_layouts/script_eks.sh:287]:
```suggestion

  if [ -n $USER_RUNNER_ID ]; then
    echo "Fix temp directories owner before cleanup ..."
    chown -R $USER_RUNNER_ID "$(pwd)/testing" || true
    chown -R $USER_RUNNER_ID "/deckhouse/testing" || true
    chown -R $USER_RUNNER_ID /tmp || true
  else
    echo "Fix temp directories permissions before cleanup ..."
    chmod -f -R 777 "$(pwd)/testing" || true
    chmod -f -R 777 "/deckhouse/testing" || true
    chmod -f -R 777 /tmp || true
  fi
```

**Review by aleksey-su — CHANGES_REQUESTED** [review]
have passed all cloud tests without exception Plus we need to make sure that in existing clusters a change in terraform will not cause a conversion to be necessary. Without that, you can't accept it.

**Comment by universal-itengineer** [comment]
e2e test from test repo
SCR-20240228-navd
SCR-20240228-nbef
SCR-20240228-pvoe

**Comment by universal-itengineer** [comment]
GCP
SCR-20240229-kird
SCR-20240229-kjok

**Comment by universal-itengineer** [comment]
Azure
image

**Comment by universal-itengineer** [comment]
vSphere
image

**Comment by universal-itengineer** [comment]
Yandex.Cloud
image
image

File: modules/040-terraform-manager/templates/terraform-auto-converger/deployment.yaml [hunk 2]
```diff
@@ -97,6 +99,12 @@ spec:
   {{- end }}
       - name: kube-rbac-proxy
         {{- include "helm_lib_module_container_security_context_read_only_root_filesystem" . | nindent 8 }}
+        #securityContext:
+        #  readOnlyRootFilesystem: true
+        #  allowPrivilegeEscalation: false
+        #  runAsGroup: 65534
+        #  runAsNonRoot: true
+        #  runAsUser: 65534
```
Comment (RomanenkoDenys) [inline modules/040-terraform-manager/templates/terraform-auto-converger/deployment.yaml:107]: why you comment it ?

**Review by RomanenkoDenys — COMMENTED** [review]
some question

File: modules/040-terraform-manager/templates/terraform-auto-converger/deployment.yaml (hunk 2 above)
Comment (universal-itengineer) [reply modules/040-terraform-manager/templates/terraform-auto-converger/deployment.yaml]: I forgot to delete it, I corrected it

**Review by RomanenkoDenys — APPROVED** [review]

**Review by aleksey-su — APPROVED** [review]

**Review by diafour — APPROVED** [review]
I've seen it before, in the testing phase, so approve from me.

**Pull request merged**